src/olympia/users/models.py

```diff
@@ -451,6 +451,51 @@ def ban_and_disable_related_content(self):
         self.delete_or_disable_related_content(delete=False)
         return self.delete(keep_fxa_id_and_email=True)
 
+    @classmethod
+    def ban_and_disable_related_content_bulk(cls, users, disable_files=False):
+        """Like ban_and_disable_related_content, but in bulk. """
+        from olympia.addons.models import Addon, AddonUser
+        from olympia.addons.tasks import index_addons
+        from olympia.bandwagon.models import Collection
+        from olympia.files.models import File
+        from olympia.ratings.models import Rating
+
+        # collect affected addons
+        addon_ids = set(
+            Addon.unfiltered.exclude(status=amo.STATUS_DELETED)
+            .filter(addonuser__user__in=users).values_list('id', flat=True))
+
+        # First addons who have other authors we aren't banning
+        addon_joint_ids = set(
+            AddonUser.objects.filter(addon_id__in=addon_ids)
+            .exclude(user__in=users).values_list('addon_id', flat=True))
+        AddonUser.objects.filter(
+            user__in=users, addon_id__in=addon_joint_ids).delete()
+
+        # Then deal with users who are the sole author
+        addons_sole = Addon.unfiltered.filter(
+            id__in=addon_ids - addon_joint_ids)
+        # set the status to disabled - using the manager update() method
+        addons_sole.update(status=amo.STATUS_DISABLED)
```

Comment:
Should we set the file status to `STATUS_DISABLED` here as well, especially if `disable_files` is false?

Reply:
We could, though `Addon.force_disable()` didn't do that - it just set the Addon status to `STATUS_DISABLED` and then `watch_disabled` function disabled the files with `File.hide_disabled_file` but didn't change the status of the `File` objects. The addon status being `STATUS_DISABLED` is sufficient to block access to it (in theory)

Reply:
Yeah, that's something I don't like about `Addon.force_disable()`. (I know that it's also not a trivial problem, there are good cases for either decision.)

I'd prefer to set `STATUS_DISABLED` on the files here as the use case is slightly different from just calling `Addon.force_disable()` (from within the Reviewer Tools for example). It also helps being consistent and avoids edge cases.

Reply:
https://github.com/mozilla/addons-server/pull/11388/commits/4ff0cba4882545731321ee8c1edeb6697d690354